src/Microsoft.AspNetCore.OData/Query/Container/TruncatedCollectionOfT.cs

```diff
@@ -29,8 +29,10 @@ public class TruncatedCollection<T> : List<T>, ITruncatedCollection, IEnumerable
         /// <param name="source">The collection to be truncated.</param>
         /// <param name="pageSize">The page size.</param>
         public TruncatedCollection(IEnumerable<T> source, int pageSize)
-            : base(source.Take(checked(pageSize + 1)))
+            : base(checked(pageSize + 1))
         {
+            var items = source.Take(checked(pageSize + 1));
```

Comment:
Would it make sense to do something like this to avoid the same calculation again (and opening it up for potential inconsistencies in the future)?
```suggestion
            var items = source.Take(Capacity);
```

Reply:
You are right, applied